### PR TITLE
Batcher Lambda, infrastructure and build

### PR DESCRIPTION
**Pull request by kenoir**
## What does this change?

Part of https://github.com/wellcomecollection/catalogue-pipeline/issues/2721

Introduce lambda infrastructure for batcher service and start setup build process.

This change does not include code changes to allow the batcher to run as a Lambda other than a skeleton class and handler function.

Depends: https://github.com/wellcomecollection/terraform-aws-ecs-service/pull/53

Next (in a new PR): Add queue & lambda trigger functionality

## How to test

- [x] Can you run a containerised build as a lambda using the [Runtime Interface Emulator](https://github.com/aws/aws-lambda-runtime-interface-emulator/)?
- [x] Does the CI process correctly build, push to ECR and deploy the Lambda container?

## How can we measure success?

When code changes are ready from https://github.com/wellcomecollection/catalogue-pipeline/issues/2721, they can be easily deployed.

## Have we considered potential risks?

We should avoid disrupting the existing infra, build & deploy and create a parallel lambda implementation to mitigate risk involved with breaking the existing process.

### CODE REVIEW

File: infrastructure/modules/scaling_service/main.tf [hunk 1]
```diff
@@ -10,7 +10,7 @@ locals {
 }
 
 module "service" {
-  source = "git::github.com/wellcomecollection/terraform-aws-ecs-service.git//modules/service?ref=v3.13.1"
+  source = "git::github.com/wellcomecollection/terraform-aws-ecs-service.git//modules/service?ref=rk_add-entrypoint"
```
Comment (paul-butcher) [inline infrastructure/modules/scaling_service/main.tf:13]: I suspect this needs to change back.

Comment (kenoir) [reply infrastructure/modules/scaling_service/main.tf:13]: Yeppers, but that branch needs merging and releasing first 👍

Comment (paul-butcher) [reply infrastructure/modules/scaling_service/main.tf:13]: There are a few more `rk_add-entrypoint`s elsewhere. I recall this was some kind of automated IDE change or similar last time.

Comment (kenoir) [reply infrastructure/modules/scaling_service/main.tf]: PR merged and released: https://github.com/wellcomecollection/terraform-aws-ecs-service/releases/tag/v4.1.0, updated module references to match.

**Review by paul-butcher — APPROVED** [review]

**Pull request merged**